### DISCUSSION TITLE
v0.2.1+8

### DISCUSSION THREAD

**Discussion by hjiangsu** [thunder-app/thunder#115] [Announcements]
## Alpha Release [v0.2.1+8]

Watch out! Theres another update incoming 🎉 

**Important: it is highly highly recommended to reinstall the app for the best (and least buggy) experience possible**

This update includes some major things under the hood to allow Thunder's functionality to work even better  and smoother than before. These under the hood changes will make it easier to implement further features in the future, so get excited for that. There's mainly a lot of quality of life changes, and features that have been suggested by the community many times!

As always, this is an alpha release, so do expect bugs to pop up here and there. Without further ado, here are the changes:

### Major Additions
- The underlying lemmy API has been migrated to use lemmy_api_client (the same one that Liftoff uses!). This is to allow for faster development of features, and to hopefully make one unified dart lemmy api client
- The underlying lemmy API change is a large change, so there may be some bugs that come from that migration. If you notice any issues that were not previously there, make sure to create a GitHub issue for it so that it can be fixed ASAP
- Addition of adaptive icons for Android. This will hopefully make Thunder stick out less like a sore thumb on Android devices - contribution from @coslu
- There is a new icon for inbox mentions and replies to mark them as "read"
- Added default sort, and feed types to Settings - contribution from @JulianPaulus
- Added Material You themes for Android users! For iOS users, there will be a feature in the future to allow you to select from a list of pre-made themes/colours
- You can now swipe across the bottom navigation bar to bring up the side menu on the feed page - no more having to press the hamburger icon on the top left

### QoL Improvements
- Settings has been re-organized to make it easier to find what you're looking for
- Adjusted the "mail" icon in the inbox to be more descriptive of what it does
- Adjusted some internal logic for fetching and caching images to bring better performance and less stuttering when scrolling through large amounts of images
- Adjusted the slide-to-dismiss image feature to be handled a bit better overall
- Tapping on a comment will now only collapse/hide the replies, and not the comment itself - contribution from @vbh
- Initial work on marking posts as read - when you tap on a post, it will be marked as read and the title/description will be greyed out to indicate that status
- There is now a setting to allow you to move the thumbnail icons to the right side
- There is now a setting to allow automatic light/dark mode depending on system settings
- Added an option in settings to open up all links in an external browser by default, without using the in-app browser
- Added ability to tap in on images within posts and comments
- Added a setting to disable swipe gestures on posts for those who prefer to not use them

### Fixes and Misc
- Fixed issue with the sort type not being respected when refreshing a feed, or switching communities
- Fixed an issue where the comment upvote/downvote would not render properly
- Fixed an issue where the bottom navigation bar was black on Android devices
- Fixed an issue where Thunder would redirect you back to the feed page after pressing back in the Settings
- Fixed an issue where posts would only show a single image even if it contained multiple

And many more changes under the hood! Have a thunderous day :D

If you have gotten this far down, these are the general major plans for the next release (this could change, so its not a definitive list):
- Introduction of user profiles to allow you to see the posts, and perhaps other information for a given user. This also would include your own profile as well, to be able to see the posts you've created, anything you've saved, comments, etc
- Marking more things as read, and being able to filter out any posts that you have already seen
- Editing of posts and comments
- Better and more options for posting
- Media download for images
- Initial support for blocking communities (maybe)

----
If you find any issues or would like to suggest features, please open up a new issue! As always, contributions are always welcomed here.

<hr /><em>This discussion was created from the release <a href='https://github.com/hjiangsu/thunder/releases/tag/'>v0.2.1+8</a>.</em>

**Comment by ghost** [comment]
i installed from the izzydroid repo, i need to uninstall and reinstall when the update appear there?

**Reply by hjiangsu** [reply]
The IzzyDroid repo should update automatically! However, I'm not sure how long it takes for it to receive the update

**Reply by seelebrn** [reply]
Thank you ! 😊

**Comment by thexYeti** [comment]
Really like the app so far! As a feature request, I'd really like to switch the comment swiping gestures to different actions, in particular, to fully collapse a thread. I don't think that's possible yet, or am I just missing it? Cheers!
